packages/web/app/src/components/ui/create-project.tsx

```diff
@@ -60,44 +60,67 @@ export const CreateProjectMutation = graphql(`
 `);
 
 const formSchema = z.object({
-  name: z.string().min(1, {
-    message: 'Project name is required',
-  }),
-  type: z.nativeEnum(ProjectType, {
-    message: 'Project type is required',
+  projectName: z
+    .string({
+      required_error: 'Project name is required',
+    })
+    .min(2, {
+      message: 'Project name must be at least 2 characters long',
+    })
+    .max(50, {
+      message: 'Project name must be at most 50 characters long',
+    })
+    .regex(
+      /^([a-z]|[0-9]|\s|\.|,|_|-|\/|&)+$/i,
+      'Project name restricted to alphanumerical characters, spaces and . , _ - / &',
+    ),
+  projectType: z.nativeEnum(ProjectType, {
+    required_error: 'Project type is required',
   }),
 });
 
-function ProjectTypeCard(props: {
+type ProjectTypeCardProps = {
   title: string;
   description: string;
   type: ProjectType;
   icon: FunctionComponentElement<{ className: string }>;
-}) {
+};
+
+export function ProjectTypeCard({
+  title,
+  description,
+  type,
+  icon,
+}: ProjectTypeCardProps): ReactElement {
   return (
     <FormItem>
-      <FormLabel className="[&:has([data-state=checked])>div]:border-primary">
+      <FormLabel className="[&:has([data-state=checked])>div]:border-primary cursor-pointer">
         <FormControl>
-          <RadioGroupItem value={props.type} className="sr-only" />
+          <RadioGroupItem value={type} className="sr-only" />
         </FormControl>
         <div className="border-muted hover:border-accent hover:bg-accent flex items-center gap-4 rounded-md border-2 p-4">
-          <Slot className="size-8 text-gray-400">{props.icon}</Slot>
+          <Slot className="size-8 text-gray-400">{icon}</Slot>
           <div>
-            <span className="text-sm font-medium">{props.title}</span>
-            <p className="text-sm text-gray-400">{props.description}</p>
+            <span className="text-sm font-medium">{title}</span>
+            <p className="text-sm text-gray-400">{description}</p>
           </div>
         </div>
       </FormLabel>
     </FormItem>
   );
 }
 
-export const CreateProjectModal = (props: {
+type CreateProjectModalProps = {
   isOpen: boolean;
   toggleModalOpen: () => void;
   organizationId: string;
-}) => {
-  const { isOpen, toggleModalOpen } = props;
+};
+
+export const CreateProjectModal = ({
+  isOpen,
+  toggleModalOpen,
+  organizationId,
+}: CreateProjectModalProps): ReactElement => {
```

Comment:
Let's please not use destructuring for the props. Accessing `props.foo` is fine!

Reply:
Fixed :)